Fix `bundlewatch` GH Action

tomalec opened the pull request:
### Changes proposed in this Pull Request:

- Use `actions/setup-node@v2` GH action,
   to use `.nvmrc` version of Node.
   :warning: It changed the `install-deps=yes`, as I don't get why we would like to force and not to use the cache. - Please, correct me if I'm wrong.
- Always use npm cache.

### Screenshots:

Restoring npm cache works for subsequent pushes:
https://github.com/woocommerce/google-listings-and-ads/runs/4396876752?check_suite_focus=true#step:3:16
![image](https://user-images.githubusercontent.com/17435/144450643-aed77227-e4f7-46bc-9f69-428be7eaf91f.png)

### Detailed test instructions:

1. See GH actions fired for this PR


<!--
Optional.
Enter a summary of all changes in this Pull Request, which will be added to the changelog if accepted.
Each line should start with `(Fix|Add|Tweak|Update) - `, for example:
> Fix - I took care of something that wasn't working.
> Add - I added something new that's pretty cool.
> Tweak - I made a small change.
> Update - I made big changes to something that wasn't broken.

Leave the "Changelog entry" header in place completely empty, without any summary if no changelog entry is needed.
If you remove the "Changelog entry" header, the title of Pull Request will be used as the changelog entry.
-->
### Changelog entry

tomalec committed:
Use `actions/setup-node@v2` GH action,

to use .nvmrc version of Node.

Always use npm cache.

tomalec committed:
Update punctuation in code comments.

tomalec committed:
Update `@wordpress/scripts@17.1.0` from `12.5.0`,

Add `postcss@8.3.5` as this is a required peer dependency.
Update webpack conf to use the latest version of Sass options.

tomalec committed:
Update package-lock.json

tomalec committed:
Whitespace alignment in php file,

to make `lint:php` happy.

eason9487 commented on .github/actions/prepare-node/action.yml:
```diff
@@ -15,27 +10,20 @@ inputs:
-      # Update npm cache directory if package-lock has changed
+      # `actions/setup-node` should update npm cache directory if `package-lock` has changed.
       run: |
-        if [ "${{ inputs.install-deps }}" = "yes" ] || [ "${{ steps.npm-cache.outputs.cache-hit }}" != "true" ]; then
```
> :warning: It changed the `install-deps=yes`, as I don't get why we would like to force and not to use the cache.

💅  or 📜 

The `install-deps` option with "maybe" value was used to handle updating the `~/.npm` cache by running `npm ci` to pre-download the package when the cache is not hit, so that subsequent jobs can directly skip the cache updates and save some time.

- Ref: https://github.com/actions/cache#skipping-steps-based-on-cache-hit
- Cache hit: https://github.com/woocommerce/google-listings-and-ads/actions/runs/1504871271
- Cache miss: https://github.com/woocommerce/google-listings-and-ads/actions/runs/1521836374

However, the constructed way of current workflows in this repo didn't bring obvious benefits in practical, since we don't have three or more layers of dependence jobs in a workflow.

Even if `actions/setup-node@v2` does [expose the `cache-hit`](https://github.com/actions/setup-node/blob/v2.5.0/action.yml#L34-L36), it does not have to be re-applied to this already deleted logic. Considering that after this logic was deleted, the original purpose of several "Setup for jobs" to pre-update `~/.npm` cache is no longer useful. Maybe we could remove all "Setup for jobs" to see if the overall time consumed make no difference, or even less.

- ["Setup for jobs" of js-css-linting](https://github.com/woocommerce/google-listings-and-ads/blob/8b42d604bdbf4a0af3ca74a363c4eec4436f16e9/.github/workflows/js-css-linting.yml#L11-L19) and `needs: Setup` in the below jobs.
- ["Setup for jobs" of php-unit-tests](https://github.com/woocommerce/google-listings-and-ads/blob/8b42d604bdbf4a0af3ca74a363c4eec4436f16e9/.github/workflows/php-unit-tests.yml#L18-L19)

tomalec replied:
> The install-deps option with "maybe" value was used to handle updating the ~/.npm cache by running npm ci to pre-download the package when the cache is not hit, so that subsequent jobs can directly skip the cache updates and save some time.

I thought, that's what `actions/setup-node@v2`'s  [`cache: 'npm'`](https://github.com/woocommerce/google-listings-and-ads/pull/1133/files#diff-bbf9b957d0e2e4f07bdbe3c523584aea587cd63068c0f45c69c1a8fb6120dcf9R17) is for:
https://github.com/actions/setup-node#caching-packages-dependencies

But it seems even though, the cache is restored, it still runs install https://github.com/woocommerce/google-listings-and-ads/runs/4396890740?check_suite_focus=true#step:3:24

I'll try using `cache-hit` explicitly

tomalec replied:
I tried using `cache-hit` to avoid doing `npm ci` but then it seems the node_modules is not there 
https://github.com/woocommerce/google-listings-and-ads/runs/4407826045?check_suite_focus=true#step:3:24

tomalec replied:
Ah, now I got it `actions/setup-node@v2` caches the `~/.npm/` cache only, not `node_modules`. So we do need to call `npm ci` if the cache is hit. I'd revert my experiments, to what was here  in OP.

When we want, to improve that we can do that in a new PR, and as you said, either 
- _"remove all "Setup for jobs" to see if the overall time consumed make no difference, or even less."_
- or make it so subsequent steps (CSS, JS linting) would not call `npm ci` at all, to make thing simple

eason9487 reviewed (approved):
LGTM. Left a non-blocking comment could do some further experiments.

TIL: `actions/setup-node@v2` has added a new `node-version-file` option to support `.nvmrc` file in the last few days. 🎉 👍  We discussed this during GHA migration, and it had to use some tricky workarounds if want to achieve this at that time. 🙈

tomalec committed:
Use `format` instead of `format-js`,

to avoid `@wordpress/scripts` deprecation warning.

Addresses https://github.com/woocommerce/google-listings-and-ads/pull/1132#pullrequestreview-822440060

tomalec committed:
Merge pull request #1132 from woocommerce/tweak/bump-wp-scripts-3

tomalec commented:
Sorry for prematurely merging https://github.com/woocommerce/google-listings-and-ads/pull/1132 to this branch, and obfuscating the diff :|

tomalec committed:
Merge branch 'develop' into fix/actions-node

tomalec commented:
Merging to unlock bundlewatch job for other PRs.
I added a note regarding [our discussion](https://github.com/woocommerce/google-listings-and-ads/pull/1133/#discussion_r761753593) about `npm ci` improvements to https://github.com/woocommerce/google-listings-and-ads/issues/1047.
